### PR TITLE
Add LinearizationType, which will be used for sequential methods.

### PR DESCRIPTION
**Pull request by atgeirr**
This consists of changes made to support a sequential solution approach. The biggest changes will be downstream in opm-simulators and not here. This PR can be merged independently, so that is what we should do. It should not change results or performance at all.

This branch is revised by me, I considered it as "including my review comments".

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
I can confirm that I get identical results with this PR. Tested on Norne with 8 processes.

File: opm/models/blackoil/blackoilpolymermodules.hh [hunk 1]
```diff
@@ -884,6 +884,7 @@ public:
         // Use log(v0) as initial value for u
         auto u = v0AbsLog;
         bool converged = false;
+        // TODO make this into parameters
```
Comment (GitPaean) [inline opm/models/blackoil/blackoilpolymermodules.hh:887]: Not clear what this comment is about.

Comment (atgeirr) [reply opm/models/blackoil/blackoilpolymermodules.hh:887]: It refers to the method parameters on the next lines.

Comment (GitPaean) [reply opm/models/blackoil/blackoilpolymermodules.hh:887]: Sorry if I am too slow on this. I do not see `method parameters` on the next lines.  Could you please just copy the lines or give a little more details? Thanks.

File: opm/models/discretization/common/linearizationtype.hh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
```
Comment (GitPaean) [inline opm/models/discretization/common/linearizationtype.hh:1]: Is it necessary to have this file in CMakeLists_files.cmake?

**Comment by GitPaean** [comment]
FYI, I could not reproduce the jenkins build failure locally. @akva2 , any suggestions?

**Comment by akva2** [comment]
it is like you point out, missing installation of the new header. jenkins uses installed builds, not super or sibling build.

**Comment by GitPaean** [comment]
> it is like you point out, missing installation of the new header. jenkins uses installed builds, not super or sibling build.

Sure, Thanks. I think both me and @atgeirr using sibling build.

**Comment by atgeirr** [comment]
> it is like you point out, missing installation of the new header

Amended and force pushed.

**Comment by atgeirr** [comment]
jenkins build this please

File: opm/models/blackoil/blackoilpolymermodules.hh (hunk 1 above)
Comment (GitPaean) [reply opm/models/blackoil/blackoilpolymermodules.hh:887]: My best guess here is to have a newton solver function as a template parameter.

**Comment by GitPaean** [comment]
It looks good to me, I have no more comments.

**Comment by atgeirr** [comment]
> It looks good to me, I have no more comments.

Thanks, I'll merge it then.

**Pull request merged**